packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -0,0 +1,123 @@
+import * as cdk from '@aws-cdk/core';
+import { TagOption } from '../common';
+import { TagUpdatesOptions } from '../constraints';
+import { IPortfolio, Portfolio } from '../portfolio';
+import { IProduct, Product } from '../product';
+import { CfnPortfolioProductAssociation, CfnResourceUpdateConstraint, CfnTagOption, CfnTagOptionAssociation } from '../servicecatalog.generated';
+import { hashValues } from './util';
+import { InputValidator } from './validation';
+
+export class AssociationManager {
+  public static associateProductWithPortfolio(scope: cdk.Resource, portfolio: IPortfolio, product: IProduct) {
+    const associationKey = hashValues(portfolio.node.addr, product.node.addr, scope.stack.node.addr);
+    if (!this.associationMap.has(associationKey)) {
+      const association = new CfnPortfolioProductAssociation(scope, `PortfolioProductAssociation${associationKey}`, {
+        portfolioId: portfolio.portfolioId,
+        productId: product.productId,
+      });
+      this.associationMap.set(associationKey, new Map());
+      this.associationMap.get(associationKey).set(Constraints.ASSOCIATION, association);
```

Comment:
Why do we do this in two steps? Why not just

```suggestion
      const associationValue = new Map<string, any>();
      associationValue.set(Constraints.ASSOCIATION, association);
      this.associationMap.set(associationKey, associationValue);
```